packages/react-router-dev/vite/cloudflare-dev-proxy.ts

```diff
@@ -43,13 +44,27 @@ export const cloudflareDevProxyVitePlugin = <Env, Cf extends CfProperties>(
 
   return {
     name: PLUGIN_NAME,
-    config: () => ({
-      ssr: {
-        resolve: {
-          externalConditions: ["workerd", "worker"],
+    config: async (userConfig) => {
+      await preloadViteEsm();
+      const vite = importViteEsmSync();
+      // a compatibility layer from Vite v6+ and below because
+      // Vite v6 overrides the default resolve.conditions, so we have to import them
+      // and if the export doesn't exist, it means that we're in Vite v5, so an empty array should be used
+      // https://vite.dev/guide/migration.html#default-value-for-resolve-conditions
+      const serverConditions =
+        userConfig.ssr?.resolve?.externalConditions ??
```

Comment:
Merging the external conditions from the user config is new behaviour that should be considered separately from this Vite 6 fix. Can we roll this back here and in the other two places?

Reply:
Oops, indeed. Rolled back.